### DISCUSSION TITLE
sed 与 grep 正则匹配不一致

### DISCUSSION THREAD

**Discussion by Dixiao-L** [physics-data/faq#96] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/41122304/131622599-29f69400-ba63-499a-9358-29dfbebc3fc8.png)
如图
`echo "1268326 4719747 992222901 - - * 164 130 " | sed -r "s/([\+-\*\/]) (-?[0-9]+) (-?[0-9]+)/foo/"`
`echo "1268326 4719747 992222901 - - * 164 130 " | grep -E "([\+-\*\/]) (-?[0-9]+) (-?[0-9]+)"`
结果不一致

**Comment by Dixiao-L** [comment, marked as answer]
sed 会将 `\/` 转义，将 sed 的运算符部分的匹配规则换成 `[\+-/\*]` 即可

**Comment by heroxbd** [comment]
`s///` 命令中的 '/' 可以换成其它字符，避免与待替换的字符串里的 `/` 冲突。例如

`echo "1268326 4719747 992222901 - - / 164 130 " | sed -r "s|([+*/-]) (-?[0-9]+) (-?[0-9]+)|foo|"`
